BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -461,19 +461,29 @@ extension ConvivaAnalytics: BitmovinPlayerListenerDelegate {
     func onAdStarted(_ event: AdStartedEvent) {
         let adPosition: AdPosition = AdEventUtil.parseAdPosition(event: event, contentDuration: player.duration)
         client.adStart(sessionKey, adStream: .ADSTREAM_SEPARATE, adPlayer: .ADPLAYER_CONTENT, adPosition: adPosition)
+        client.detachPlayer(sessionKey)
```

Comment:
You mentioned `to detach and reattach the player on ad breaks.` in the PR description, however, you are using the `onAdStarted` listener here 🤔 `AdStarted` != `AdBreakStarted`.

Not sure if this could be a problem if we `attach` and `detach` the player right after each other (in case there are multiple `Ads` in one `AdBreak`)

Reply:
Good point, and if we use `AdBreakStarted` and `AdBreakEnded`, there should also be no case where the need to pay attention to errored ads, skipped ads etc unlike `AdFinished`).

Reply:
Attaching and detaching the player shouldn't be a problem.
AdBreakStarted and AdBreakEnded aren't functions currently available on the BitmovinPlayerListenerDelegate. If those can get added then I agree that it makes more sense to use those.

Reply:
Should be fairly easy to add 🙂  but we can also implement this on our side first if you prefer that.

Reply:
@austinchristensen , @dweinber , @stonko1994 , it seems that there are 2 ways of monitoring Ads in Conviva. I am not sure if these are complementary or exclusive.
- Conviva Ad Breaks (monitor ad break metadata) 
- Conviva Ad Experience (create ad monitoring sessions)

Without more insights into how they differ, my rough understanding is that Conviva recommends attach/detach player for Ad Experience (https://community.conviva.com/site/global/platforms/other_platforms/sdks/ios_sdk/tutorials/integratecustom/old-index.gsp#handling_ads)

So we need to understand the difference between the 2 approaches and confirm if detaching/attaching player for AdBreaks instead of Ads would suffice Conviva requirements before making such change.

Reply:
Those 2 ways of ad tracking are exclusive. We implemented that back in the days for the web integration: https://github.com/bitmovin/bitmovin-player-analytics-conviva/pull/22. We introduce a config (`AdTrackingMode`) to make this configurable.

We `detach` / `attach` it only in the `AdExperienceTrackingPlugin.ts`. IIRC `detach` and `attach` is not needed for `Ad Break Tracking` as we do it on iOS 🤔

Reply:
@dweinber Good call on using ad break start and ad finished. We've updated it to use those.
@luckygoyal-bitmovin We've reviewed the two different approaches to ad tracking. This PR should sufficient to fix this bug for Ad Breaks. We are reviewing implementing Ad Experience.

Reply:
I think that's sufficient for the customer so let's keep this solution. I don't think there's a new to implement Ad Experience at the moment (but @luckygoyal-bitmovin to confirm).

Reply:
@austinchristensen , thanks for confirming that this solution should be sufficient for now. I agree with @dweinber on Ad experience implementation, we will make a separate request as and when the Ad experience requirement is raised by the customer.